google/cloud/bigquery/table.py

```diff
@@ -452,17 +453,20 @@ def schema(self):
                 instance or a compatible mapping representation of the field.
         """
         prop = self._properties.get(self._PROPERTY_TO_API_FIELD["schema"])
-        if not prop:
+        if not prop:  # if empty Schema, empty list, None
             return []
-        else:
-            return _parse_schema_resource(prop)
+        elif isinstance(prop, Schema):
+            return prop
+        return _parse_schema_resource(prop)
```

Comment:
Presumably we want to return a `Schema` object now. 

```suggestion
        return Schema.from_api_repr(prop)
```

Aside: Is `_parse_schema_resource` dead code? I assume it still returns a list of `SchemaField`?

Reply:
`_parse_schema_resource` still returns a list of `SchemaFields` for backwards compatibility.

Where possible I *-tried-* to stay consist about what the user provides versus gets back: if someone handed the library a `list` of `SchemaField`s (i.e. cause that is what they are used to) then we maintained that state. I tried to avoid randomly converting `lists` into `Schema` objects to avoid user surprise.